source/libs/opl.h

```diff
@@ -2653,39 +2727,29 @@ int opl_loadbank_op2(opl_t* opl, void const* data, int size ) {
     return(-3);
   }
   buff += 8;
+  
+  opl->is_op2 = true;
 
   /* load 128 instruments from the IBK file */
   for (i = 0; i < 175; i++) {
     /* load instruments */
-    int n = i;
-    if( i >= 128 ) {
-        n = 35 + i;
-    }
-
-    /* load modulator */
-    opl->opl_gmtimbres[n].modulator_E862 = buff[7]; /* wave select */
-    opl->opl_gmtimbres[n].modulator_E862 <<= 8;
-    opl->opl_gmtimbres[n].modulator_E862 |= buff[6]; /* sust/release */
-    opl->opl_gmtimbres[n].modulator_E862 <<= 8;
-    opl->opl_gmtimbres[n].modulator_E862 |= buff[5]; /* attack/decay */
-    opl->opl_gmtimbres[n].modulator_E862 <<= 8;
-    opl->opl_gmtimbres[n].modulator_E862 |= buff[4]; /* AM/VIB... flags */
-    /* load carrier */
-    opl->opl_gmtimbres[n].carrier_E862 = buff[14]; /* wave select */
-    opl->opl_gmtimbres[n].carrier_E862 <<= 8;
-    opl->opl_gmtimbres[n].carrier_E862 |= buff[13]; /* sust/release */
-    opl->opl_gmtimbres[n].carrier_E862 <<= 8;
-    opl->opl_gmtimbres[n].carrier_E862 |= buff[12]; /* attack/decay */
-    opl->opl_gmtimbres[n].carrier_E862 <<= 8;
-    opl->opl_gmtimbres[n].carrier_E862 |= buff[11]; /* AM/VIB... flags */
-    /* load KSL */
-    opl->opl_gmtimbres[n].modulator_40 = ( buff[9] & 0x3f ) | ( buff[8] & 0xc0 );
-    opl->opl_gmtimbres[n].carrier_40 = ( buff[16] & 0x3f ) | ( buff[15] & 0xc0 );
-    /* feedconn & finetune */
-    opl->opl_gmtimbres[i].feedconn = buff[10];
-    //opl->opl_gmtimbres[n].finetune = 0;
-    opl->opl_gmtimbres[n].notenum = buff[3];
-    buff += 36;
+    
+    /* OP2 instrument header */
+    opl->op2_flags[i] = buff[0] | ((uint16_t)buff[1] << 8);
```

Comment:
This line gives an error when compiling the code as c++, adding a cast should fix it
```suggestion
    opl->op2_flags[i] = (enum op2_flags_t)( buff[0] | ((uint16_t)buff[1] << 8) );
```